src/providers/definitions.rs

```diff
@@ -26,10 +26,9 @@ pub struct SongMetadata {
 
 #[async_trait]
 pub trait Provide {
-    type SearchResultType;
-
-    async fn check(info: &SongMetadata) -> Result<()>;
-    async fn track(search_result: Self::SearchResultType) -> Result<()>;
+    /// Search a audio similar with info from Provider,
+    /// and return the audio link
+    async fn check(&self, info: &SongMetadata) -> Result<Option<String>>;
```

Comment:
@pan93412 we should not only return an audio URL but also the request settings, such as header.
https://github.com/UnblockNeteaseMusic/server/blob/enhanced/src/server.js#L130-L133

But we need to keep the code similar to the js version currently.